cuda/extensions.bzl

```diff
@@ -0,0 +1,8 @@
+"""Entry point for extensions used by bzlmod."""
+
+load("//cuda:repositories.bzl", "rules_cuda_dependencies")
+
+def _init(module_ctx):
+    rules_cuda_dependencies()
```

Comment:
Blindly loading all your transitive dependencies in a module extension is a terrible idea. Because all of them will be "internal dependencies" of rules_cuda, and may introduce duplicated dependencies in the whole dependency graph.
@cloudhan @jsharpe This is what should be eliminated.

Reply:
@meteorcloudy my mistake is so obvious now, but after having taken a short break from writing bzlmod code I'd misremembered the migration path of being able to add code in `WORKSPACE.bzlmod` vs putting it in the init function of a module. 

I suspect this might be a common mistake during migration. So any warnings that can be pulled out would be very useful, or if not, a 'lint' mode that searches for these conflicts would help.